src/operations/execute_operation.ts

```diff
@@ -188,111 +164,113 @@ function executeWithServerSelection<TResult>(
     selector = readPreference;
   }
 
-  const serverSelectionOptions = { session };
-  function retryOperation(originalError: MongoError) {
-    const isWriteOperation = operation.hasAspect(Aspect.WRITE_OPERATION);
-    const isReadOperation = operation.hasAspect(Aspect.READ_OPERATION);
+  const server = await topology.selectServerAsync(selector, { session });
 
-    if (isWriteOperation && originalError.code === MMAPv1_RETRY_WRITES_ERROR_CODE) {
-      return callback(
-        new MongoServerError({
-          message: MMAPv1_RETRY_WRITES_ERROR_MESSAGE,
-          errmsg: MMAPv1_RETRY_WRITES_ERROR_MESSAGE,
-          originalError
-        })
-      );
+  if (session == null || !operation.hasAspect(Aspect.RETRYABLE)) {
```

Comment:
I wouldn't be opposed to breaking this if statement into two - one to handle the null session case and another to handle non-retryable operations.  It would be roughly the same code in both except the null session block wouldn't have the end sessions logic.  

It's just preference though so this is optional

Reply:
done